### ISSUE TITLE
Occoquan (spec-act team): PL3_5250_0001

### ISSUE THREAD

**Issue by rburghol** [HARPgroup/HARParchive#498]
### Tasks
- [x] run Land, river
- [ ] Investigate withdrawals in Occoquan Reservoir
  - [ ] Does lack of withdrawal account for the difference between flow out of 0001 in HSPF versus HSP2?
  - [ ] how to test? Are the withdrawals getting added to the H5 file in the step import_uci?
  - [ ] Do we need to tell the run_river.csh routine to preserve the H5 so that we can manually examine it?

#### Running hsp2 on Occoquan (PL3_5250_0001):**

Running the meteorology (if needed):
```
cd /opt/model/p53/p532c-sova/
segs=`cbp get_landsegs PL3_5250_0001`
for i in $segs; do
  # convert grid CSVs into land segment CSVs
  a2l_one 1984010100 2020123123 /backup/meteorology/out/grid_met_csv /backup/meteorology/out/lseg_csv $i
  # update long term averages
  LongTermAvgRNMax /backup/meteorology/out/lseg_csv/1984010100-2020123123 /backup/meteorology/out/lseg_csv/RNMax 1 $i
  # finally, create a WDM for each land seg
  # this script reads the file /etc/hspf.config to get directories.
  wdm_pm_one $i 1984010100 2020123123 nldas2 harp2021 nldas1221 p20211221
done
```
Commands in order:
in: cd /opt/model/p53/p532c-sova/

`cbp run_lug.csh p532sova_2021 PL3_5250_0001`

`cbp run_land.csh p532sova_2021 PL3_5250_0001`

`cbp run_etm.csh hsp2_2022 PL3_5250_0001`
* Instead of p532sova_2021 in order to fix an error with eos files

`cbp run_rug.csh hsp2_2022 PL3_5250_0001`

`cbp run_river.csh hsp2_2022 PL3_5250_0001`

**Comment by juliabruneau** [comment]
### 1st try

Error with first command run_lug:
```
juliasb@deq2:/opt/model/p53/p532c-sova$ cbp run_lug.csh p532sova_2021 PL3_5250_0001
Loading /opt/model/p53/p532c-sova/hspf.config
../../..//config/seglists/PL3_5250_0001.land: No such file or directory.
../../..//config/seglists/PL3_5250_0001.land: No such file or directory.
Warning: could not find /opt/model/p53/p532c-sova/run_bhatt in tree
```

**Comment by juliabruneau** [comment]
### 2nd try

1. Ran `cbp basingen.csh hsp2_2022 5250`
2. Ran the meteorology (commands above)
3. Ran **run_lug** successfully
4. Ran **run_land** successfully
5. Ran **run_etm** unsuccessfully: 
```
juliasb@deq2:/opt/model/p53/p532c-sova$ cbp run_etm.csh p532sova_2021 PL3_5250_0001
Loading /opt/model/p53/p532c-sova/hspf.config
Trying: p532sova_2021, PL2_4970_5250, [random.ksh] | ../../..//code/bin/make_binary_transfer_coeffs.exe
 Binary ETM file for PL2_4970_5250 p532sova_2021
   reading data files
   all data read, make binary etm file
A51059 for

 PROBLEM FILE WRITTEN

 Error: no non-zero values found for annual
 max quickflow in returnfq subroutine
../../../tmp/wdm/land/for/p532sova_2021/forA51059.wdm

Trying: p532sova_2021, PL2_4970_5250, [random.ksh] | ../../..//code/bin/make_binary_transfer_coeffs.exe
 Binary ETM file for PL2_4970_5250 p532sova_2021
   reading data files
   all data read, make binary etm file
A51059 for

 PROBLEM FILE WRITTEN

 Error: no non-zero values found for annual
 max quickflow in returnfq subroutine
../../../tmp/wdm/land/for/p532sova_2021/forA51059.wdm

Warning: could not find /opt/model/p53/p532c-sova/run_bhatt in tree
```

**Comment by rburghol** [comment]
I have never seen such an error! We gotta call gopal! 

But! How about we run the hsp2_2022 land and analyze the outputs?

**Comment by juliabruneau** [comment]
> I have never seen such an error! We gotta call gopal!
> 
> But! How about we run the hsp2_2022 land and analyze the outputs?

Do you mean: `cbp run_land.csh hsp2_2022 PL3_5250_0001`?

**Comment by rburghol** [comment]
Sorry I updated that comment to auggest looking at the forest wdm with error. 

BTW: error codes;
- -81 means DSN not found 

See codes here: https://github.com/HARPgroup/cbp_wsm/blob/091af49aec3403127bb6d60845e61145b2ed030e/code/src/hspf/lib3.2/src/wdm/wdsptm.f#L44

**Comment by juliabruneau** [comment]
**Correction to the problem file**

I looked into the wrong directory for the problem file:
```
juliasb@deq2:/opt/model/p53/p532c-sova$ head problem

 PROBLEM FILE WRITTEN

forA51059.wdm
 does not exist
```
But after inspection, there _is_ a wdm file:
```
juliasb@deq2:/opt/model/p53/p532c-sova/tmp/wdm/land/for/p532sova_2021$ ls
error.fil       forA47163.wdm  forA51107.wdm  forA51595.wdm  forB51141.wdm
for.txt         forA51003.wdm  forA51111.wdm  forA51600.wdm  forB51161.wdm
for51800.1.txt  forA51011.wdm  forA51117.wdm  forA51640.wdm  forB51167.wdm
for51800.2.txt  forA51019.wdm  forA51121.wdm  forA51683.wdm  forB51169.wdm
for51800.3.txt  forA51021.wdm  forA51125.wdm  forA51685.wdm  forB51173.wdm
for51800.txt    forA51023.wdm  forA51141.wdm  forA51690.wdm  forB51185.wdm
forA37001.wdm   forA51025.wdm  forA51143.wdm  forA51720.wdm  forB51191.wdm
forA37005.wdm   forA51031.wdm  forA51147.wdm  forA51750.wdm  forB51195.wdm
forA37009.wdm   forA51035.wdm  forA51153.wdm  forA51770.wdm  forB51197.wdm
forA37033.wdm   forA51037.wdm  forA51155.wdm  forA51775.wdm  forC51071.wdm
forA37067.wdm   forA51045.wdm  forA51161.wdm  forA51800.wdm  forF51019.wdm
forA37077.wdm   forA51051.wdm  forA51167.wdm  forA54063.wdm  forF51059.wdm
forA37081.wdm   forA51059.wdm  forA51169.wdm  forB37005.wdm  forF51061.wdm
forA37135.wdm   forA51061.wdm  forA51173.wdm  forB37009.wdm  forF51107.wdm
forA37145.wdm   forA51063.wdm  forA51175.wdm  forB51003.wdm  forF51125.wdm
forA37157.wdm   forA51067.wdm  forA51185.wdm  forB51019.wdm  forF51153.wdm
forA37169.wdm   forA51071.wdm  forA51191.wdm  forB51023.wdm  forF54063.wdm
forA37171.wdm   forA51077.wdm  forA51195.wdm  forB51067.wdm  problem
forA37181.wdm   forA51081.wdm  forA51197.wdm  forB51071.wdm
forA37185.wdm   forA51083.wdm  forA51515.wdm  forB51077.wdm
forA37189.wdm   forA51089.wdm  forA51520.wdm  forB51105.wdm
forA47091.wdm   forA51105.wdm  forA51590.wdm  forB51125.wdm
juliasb@deq2:/opt/model/p53/p532c-sova/tmp/wdm/land/for/p532sova_2021$ head forA51059.wdm
#▒▒a▒a▒ei"jSkTlm!nUo q!s"u-w▒20041015132243  20041015132243  SURO▒BLANK   SURO    PERLND  surface water in/ac/ivld                       ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
```

**Comment by glenncampagna** [comment]
### Problem trying to run wdm2text
Example: 
```
/opt/model/p53/p532c-sova/tmp/wdm/land/for/p532sova_2021$ wdm2text
wdm name, start year, end year, dsn
forA51059.wdm 1984 2021 1000
```

Opening the new problem file:
```
PROBLEM FILE WRITTEN
error getting start and end of existing data                   
 error code=         -81                                        
 dsn =         1000      
```        
Maybe this means there is no data for that DSN?
Where can we find a land seg uci to become familiar with the DSNs?

**Comment by rburghol** [comment]
- First off, try making the parameters to wdm2text comma separate instead of space separated.
- Land seg UCI -- here! (we should make a script to locate those!): http://deq1.bse.vt.edu:81/p532/tmp/uci/land/for/p532sova_2021/ is the directory with all of those forest WDMs for that scenario.
-

**Comment by glenncampagna** [comment]
### Update
I was able to get wdm2text to work with the commas in between arguments and got the output:
```
wdm name, start year, end year, dsn
forA51059.wdm, 1984, 2021, 111
At line 282 of file dsn_utils.f
Fortran runtime error: End of record
Error termination. Backtrace:
#0  0x7f3862a38d21 in ???
#1  0x7f3862a39869 in ???
#2  0x7f3862a3a54f in ???
#3  0x7f3862c7ce10 in ???
#4  0x7f3862c8aeeb in ???
#5  0x7f3862c8bf76 in ???
#6  0x7f3862c7d53d in ???
#7  0x55b06dc9f93f in ???
#8  0x55b06dc99752 in ???
#9  0x55b06dc9a474 in ???
#10  0x7f38626e00b2 in ???
#11  0x55b06dc9937d in ???
#12  0xffffffffffffffff in ???

```
Also, I used 111 as the DSN because looking at the UCI, DSN 1000 corresponds to Evaporation while DSN 111 corresponds to the SURO column of PWATER (aka runoff) which you mentioned earlier in this issue. Let me know if I'm going in the wrong direction with the DSNs, but this was the first time I ran wdm2text and a problem file wasn't written

**Comment by rburghol** [comment]
I think your approach is correct in that 111 is potentially illuminating.  That said, the DSN 1000 IS something that the land simulation would read, and if that is the error source, we should start there.  But looks like none had what we really want.

BUT -- right off the bat I do see an error in your wdm2text command: you asked for data through 2021, and the model only runs till the end of 2020, so change back to 2020 and that should at least output data (tho maybe zeroes if what the model tells us is true).  And export both 111 and 1000 so that we can see what they yield.

**Comment by juliabruneau** [comment]
Running wdm2text:

With DSN 1000:
```
juliasb@deq2:/opt/model/p53/p532c-sova/tmp/wdm/land/for/p532sova_2021$ wdm2text
  program to write hourly ascii output from a wdm
 wdm name, start year, end year, dsn
forA51059.wdm, 1984, 2020, 1000
```
... seems like there was no result / no file appeared for it after looking at the files in that directory

With DSN 111:
```
juliasb@deq2:/opt/model/p53/p532c-sova/tmp/wdm/land/for/p532sova_2021$ wdm2text
  program to write hourly ascii output from a wdm
 wdm name, start year, end year, dsn
forA51059.wdm, 1984, 2020, 111
 hourly average =    0.0000000000000000
 annual average =    0.0000000000000000
```
... a file: `forA51059_0111.csv` was created:
```
juliasb@deq2:/opt/model/p53/p532c-sova/tmp/wdm/land/for/p532sova_2021$ head forA51059_0111.csv
        1984 ,           1 ,           1 ,           0 ,   0.00000000
        1984 ,           1 ,           1 ,           1 ,   0.00000000
        1984 ,           1 ,           1 ,           2 ,   0.00000000
        1984 ,           1 ,           1 ,           3 ,   0.00000000
        1984 ,           1 ,           1 ,           4 ,   0.00000000
        1984 ,           1 ,           1 ,           5 ,   0.00000000
        1984 ,           1 ,           1 ,           6 ,   0.00000000
        1984 ,           1 ,           1 ,           7 ,   0.00000000
        1984 ,           1 ,           1 ,           8 ,   0.00000000
        1984 ,           1 ,           1 ,           9 ,   0.00000000
```

**Comment by rburghol** [comment]
OK - this tells us that these things are empty (just what the model told us!).  I did some looking and found that there are some funky land segments in phase 5 that begin with "F" and we don't have any of those in our shapefiles, but they arereferenced in phase 5.3.2.  So.  I developed a process for fixing these.  It is outlined here:https://github.com/HARPgroup/HARParchive/issues/146#issuecomment-1217106686 

I fixed them already, so re-running the meteorology and then rerunning the land *should* fix it. Keep me posted tomorrow.  But, the process is there in that issue if you should ever need to refer to it.

OF course, this reminds me that I need to get the phase 6 implementation of hsp2 river simulation and ETM done, and then we can run Occoquan as Phase 6, not Phase 5, which is *really* what we should be doing.

**Comment by juliabruneau** [comment]
**Trying to run again:**

I got all the way to run_etm, and received the same error as before:
```
juliasb@deq2:/opt/model/p53/p532c-sova$ cbp run_etm.csh p532sova_2021 PL3_5250_0001
Loading /opt/model/p53/p532c-sova/hspf.config
Trying: p532sova_2021, PL2_4970_5250, [random.ksh] | ../../..//code/bin/make_binary_transfer_coeffs.exe
 Binary ETM file for PL2_4970_5250 p532sova_2021
   reading data files
   all data read, make binary etm file
A51059 for

 PROBLEM FILE WRITTEN

 Error: no non-zero values found for annual
 max quickflow in returnfq subroutine
../../../tmp/wdm/land/for/p532sova_2021/forA51059.wdm

Trying: p532sova_2021, PL2_4970_5250, [random.ksh] | ../../..//code/bin/make_binary_transfer_coeffs.exe
 Binary ETM file for PL2_4970_5250 p532sova_2021
   reading data files
   all data read, make binary etm file
A51059 for

 PROBLEM FILE WRITTEN

 Error: no non-zero values found for annual
 max quickflow in returnfq subroutine
../../../tmp/wdm/land/for/p532sova_2021/forA51059.wdm

Warning: could not find /opt/model/p53/p532c-sova/run_bhatt in tree
```
Problem file:
```
 PROBLEM FILE WRITTEN

forA51059.wdm
 does not exist
```

**Comment by rburghol** [comment]
Hey @juliabruneau @glenncampagna -- I wasn't fully clear, I think now you can rerun the meteorology step.  Did you do that before rerunning the land?  All I did was set up the files needed for the WDM creation, but did not actually create it myself.  Sorry to be vague.

```
segs=`cbp get_landsegs PL3_5250_0001`
for i in $segs; do
  # convert grid CSVs into land segment CSVs
  ./a2l_one 1984010100 2020123123 /backup/meteorology/out/grid_met_csv /backup/meteorology/out/lseg_csv $i
  # update long term averages
  LongTermAvgRNMax /backup/meteorology/out/lseg_csv/1984010100-2020123123 /backup/meteorology/out/lseg_csv/RNMax 1 $i
  # finally, create a WDM for each land seg
  # this script reads the file /etc/hspf.config to get directories.
  ./wdm_pm_one $i 1984010100 2020123123 nldas2 harp2021 nldas1221 p20211221
done
```

**Comment by juliabruneau** [comment]
> Hey @juliabruneau @glenncampagna -- I wasn't fully clear, I think now you can rerun the meteorology step. Did you do that before rerunning the land? All I did was set up the files needed for the WDM creation, but did not actually create it myself. Sorry to be vague.
> 
We did run the meteorology first and all the run_*** steps afterwards.

**Comment by rburghol** [comment]
> We did run the meteorology first and the run_*** steps afterwards.
@juliabruneau @glenncampagna 
Bummer. OK. My thought process:
- It is clear from the ETM error that the forest  A51059 didn't have any outflow, which implies that the meteorology input is zero for it.
- I looked into the meteorology, and no files exist for that land segment, implying that the met routine didn't run, or rather, didn't run *successfully* for it.  
- So, I rerun that one manually, instead of in the `do` loop to see what's up:
```
 cd /backup/meteorology/
./a2l_one 1984010100 2020123123 /backup/meteorology/out/grid_met_csv /backup/meteorology/out/lseg_csv A51059

LongTermAvgRNMax /backup/meteorology/out/lseg_csv/1984010100-2020123123 /backup/meteorology/out/lseg_csv/RNMax 1 A51059

A51059
File opened: /backup/meteorology/out/lseg_csv/1984010100-2020123123/A51059.RAD
File opened: /backup/meteorology/out/lseg_csv/RNMax/A51059.RNMax
```
- The above said `File opened: /backup/meteorology/out/lseg_csv/1984010100-2020123123/A51059.RAD` -- which is very good news!
- The last command tossed a bunch of data out, including non-zero precip values like so: ` HPRC, A51059,        1984 ,   41.8050156`, which is even better news

OK, so it definitely worked for me to run that, and now I suspect that the run_lug, run_land, run_etm sequence should succeed for that one.  Makes me wonder if:
- You all are running into a permission problem?  Shouldn't be the case but...
- Did you change directories with `cd /backup/meteorology/` before running the met loop?

**Comment by juliabruneau** [comment]
> 
Did you change directories with cd /backup/meteorology/ before running the met loop?
>
Yes, we changed to /backup/meteorology/ when running the meteorology. 

I will try running from run_lug forward and see if everything works now!

**Comment by rburghol** [comment]
Totally strange -- I am sure you did since updated met files all existed! Except A51059... which is so strange.  All good though. FYI -- a way to check on a met file to see if you need to run `a2l_one` is: 
```
Rscript /opt/model/model_meteorology/R/get_lseg_summary_stats.R A51059
cat /backup/meteorology/A51059SummaryStats.csv
```
IF the script fails, or the summary stats are empty, need to rerun. We need a more efficient way to handle this check of course, possibly using REST, but this is at least a method to gain some info.

**Comment by juliabruneau** [comment]
### Success

After changing the scenario in run_etm to hsp2_2022 (since we got the same error as with Meherrin), we were able to produce the csvs:
```
-rw-rw-r-- 1 7184376 allmodelers  73M Aug 17 18:35 PL2_4970_5250_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.2M Aug 17 18:35 PL2_4970_5250_rovol.csv
-rw-rw-r-- 1 7184376 allmodelers  71M Aug 17 18:37 PL0_5141_5140_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.0M Aug 17 18:37 PL0_5141_5140_rovol.csv
-rw-rw-r-- 1 7184376 allmodelers  73M Aug 17 18:39 PL1_5370_5470_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.2M Aug 17 18:39 PL1_5370_5470_rovol.csv
-rw-rw-r-- 1 7184376 allmodelers  73M Aug 17 18:41 PL2_5140_5360_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.2M Aug 17 18:41 PL2_5140_5360_rovol.csv
-rw-rw-r-- 1 7184376 allmodelers  72M Aug 17 18:43 PL2_5470_5360_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.2M Aug 17 18:43 PL2_5470_5360_rovol.csv
-rw-rw-r-- 1 7184376 allmodelers  68M Aug 17 18:45 PL3_5360_5250_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.2M Aug 17 18:45 PL3_5360_5250_rovol.csv
-rw-rw-r-- 1 7184376 allmodelers  73M Aug 17 18:47 PL3_5250_0001_hydr.csv
-rw-rw-r-- 1 7184376 allmodelers 9.2M Aug 17 18:47 PL3_5250_0001_rovol.csv
juliasb@deq2:/media/model/p532/out/river/hsp2_2022/hydr$ date
Wed Aug 17 18:50:13 UTC 2022
```

**Comment by rburghol** [comment]
This is fantastic! Now I think we have to determine if we need to run ETM for both scenarios, or just one?  The answer to that may tell us how important it is for us to step up to the next phase of overhauling ETM of course :)

**Comment by glenncampagna** [comment]
### Comparing hspf and hsp2 for PL3_5250_0001
`"Avg % difference: 35.4735489"`
1. 
![image](https://user-images.githubusercontent.com/104520197/185413498-a7361a63-253f-4074-a032-016d672396b7.png)

2. 
![image](https://user-images.githubusercontent.com/104520197/185413683-069329da-d5c0-49ec-a7c0-e053e3ab7bb9.png)


Our comparison script showed pretty substantial differences for this segment, the highest we've seen

**Comment by rburghol** [comment]
This one looks like there's a problem with flow routing. Just for grins, can we also redo the one in the Roanoke that had the tiny error?

**Comment by rburghol** [comment]
And also find and analyze one of the headwaters in this Occoquan watershed.

cbp get_riversegs PLxxxxxxx 

Is your friend.

**Comment by glenncampagna** [comment]
### Comparing hspf and hsp2 for PL1_5370_5470 (Occoquan headwater)
`"Avg % difference: 10.0078877"`
1. 
![image](https://user-images.githubusercontent.com/104520197/185416739-4ddd5af0-480f-44b0-bdde-3f1aa1be989d.png)

2. 
![image](https://user-images.githubusercontent.com/104520197/185417305-edd64a2a-6c9f-48a8-ba41-2773846cbcd8.png)

**Comment by juliabruneau** [comment]
> This one looks like there's a problem with flow routing. Just for grins, can we also redo the one in the Roanoke that had the tiny error?

We just ran the river seg OR1_7700_7980 again and received the same avg %-diff of 0.000619 %.

**Comment by rburghol** [comment]
Allrighty - this is interesting. Does it actually *look* like a 10% error in PL1_5370_5470??

Also, to test out the routing issue/question, can you summarize flows (mean Q, % error) for the next segment downstream of PL1_5370_5470, and for the 2 upstream segments that flow directly into PL3_5250_0001?  I want to see if one of the branches flowing into 5250 is wonky but maybe the nother is OK?

Thanks!

**Comment by rburghol** [comment]
Allrighty - this is interesting. Does it actually *look* like a 10% error in PL1_5370_5470??

Also, to test out the routing issue/question, can you summarize flows (mean Q, % error) for the next segment downstream of PL1_5370_5470, and for the 2 upstream segments that flow directly into PL3_5250_0001?  I want to see if one of the branches flowing into 5250 is wonky but maybe the nother is OK?

Thanks!

**Comment by glenncampagna** [comment]
### Summarizing flows - in progress
- For 2 upstream segments that flow directly into PL3_5250_0001:
1st seg: _PL3_5360_5250_
  for hsp2: `"Avg Q: 373.447964237367 cfs"`
  for hspf: `"Avg Q: 366.455165376594 cfs"`

**Comment by rburghol** [comment]
@glenncampagna @juliabruneau this is good info -- and encouraging as it at least seems that our method of passing flow from one watershed downstream is working (in most cases).  This is so long we should probably meet up to plot next steps.  But for background before we meet, here are some observations/questions:
- Inflow to PL3_5250_0001 should be nearly 600 cfs regardless of the model used.  Can you use the wdm2text routine to compare the inflow for hsp2 and hspf?
- How do we generate summaries of the HSPF models?  I don't think we added that to the task list (I should have!), but do you all have a method to generate stats? I should add them into `run_river.csh`

One thing that seems to be occurring is the difference between "mean daily flow error", and "error in mean flow".  In other words, the method of calculation matters:
- mean daily flow error: calculate the error % each day (> 10,000 days), then find the mean of these numbers
- error in mean flow: calculate the mean values for flow over the whole period, then calculate a % difference between these 2 numbers

Neither of these methods is right or wrong of course, but they will have different results. 
- Ex: 199.37 cfs and 198.6 cfs
  - mean daily flow error: 3.994%
  - error in mean flow: 0.39%
- Ex: 373.44 and 366.46
  - mean daily error: 12.3%
  - error in mean flow: 1.9%

Though both are "correct" in a sense, the 1st method can exaggerate errors due to time step lagging -- something we *think* we resolved, but should be aware of, and also, when we compare a model to a stream gage for example, we almost always accept that a lag may occur :).  So, even if we calculate a metric or two with the 1st method, the most commonly used method is the 2nd method -- it is that method that is essentially what I am using to make the comparisons that I referenced in the Rockfish issue in a recent comment -- there was a bunch of comments in a row and it may have been buried, but I noted that the flow alteration stats that your script calculated (L90_Qout, L30_Qout, ...) were identical down to like 5 decimal places.  See links here: https://github.com/HARPgroup/HARParchive/issues/443#issuecomment-1219737276

**Comment by glenncampagna** [comment]
### Comparing inflows to PL3_5250_0001 using wdm2text
Inflow corresponds to DSN 11 (IVOL)
For hspf: 
```
wdm name, start year, end year, dsn
PL3_5250_0001.wdm, 1984, 2020, 11
 hourly average =    50.026447587462819     
 annual average =    438556.17674241727
```
For hsp2:
```
wdm name, start year, end year, dsn
PL3_5250_0001.wdm, 1984, 2020, 11
 hourly average =    50.590926827928094     
 annual average =    443504.67637585831
```
IVOL is originally in units of ac-ft/hr , so with the conversion to cfs ( x 12.1) these Qout values are approximately 605 cfs each

**Comment by glenncampagna** [comment]
### Qout comparison for PL3_5250_0001  and corrected error in mean flow
Used: `Rscript HARParchive/HARP-2022-Summer/compare_hspf_hsp2.R PL3_5250_0001 /media/model/p532/out/river/p532sova_2021/stream/PL3_5250_0001_0111.csv /media/model/p532/out/river/hsp2_2022/hydr/PL3_5250_0001_hydr.csv`

Output:
```
[1] "Avg Q for hspf: 557.02 cfs"
[1] "Avg Q for hsp2: 614.9 cfs"
[1] "% error in mean flow:  9.878"
```

**Comment by glenncampagna** [comment]
It seems like DIVA (3008) and DIVR (3007) are not present in both the hspf and hsp2 wdm, since I got an updated problem file similar to this for all 4 `wdm2text` runs:
```
PROBLEM FILE WRITTEN
 
 error getting start and end of existing data                   
 error code=         -81                                        
 dsn =         3007 

```
Your question of whether or not hsp2 has withdrawals or not makes sense though, as that would've explained the ~55 cfs difference between the 2 Qout values, but this testing seems to show that neither hsp2 of hspf has diversions?

**Comment by rburghol** [comment]
Ah!  
1. They are `WDM3` in the `EXT SOURCES` block, which means they are in the file `ps_sep_div_ams_hsp2_2022_PL3_5250_0001.wdm`: `WDM3       23   ps_sep_div_ams_hsp2_2022_PL3_5250_0001.wdm`
2. Can we look into the h5 file for those ?
